ledger/acctupdates_test.go

```diff
@@ -596,3 +597,157 @@ func TestLargeAccountCountCatchpointGeneration(t *testing.T) {
 		}
 	}
 }
+
+// The TestAcctUpdatesUpdatesCorrectness conduct a correctless test for the accounts update in the following way -
+// Each account is initialized with 100 algos.
+// On every round, each account move variable amount of funds to an accumulating account.
+// The deltas for each accounts are picked by using the lookup method.
+// At the end of the test, we verify that each account has the expected amount of algos.
+// In addition, throughout the test, we check ( using lookup ) that the historical balances, *beyond* the
+// lookback are generating either an error, or returning the correct amount.
+func TestAcctUpdatesUpdatesCorrectness(t *testing.T) {
```

Comment:
does it constantly reproduce the bug?

Reply:
On my laptop, this unit test consistently reproducing the bug. Note that it's using scenarios which aren't exercised in algod - i.e. looking back beyond the loopback window, having a very short loopback window to allow changes to buffer, etc.